src/pages/index.js

```diff
@@ -1,94 +1,217 @@
 
 function handleAddCardSubmit(data) {
-  renderCard(data);
-  addCardPopup.close();
+  addCardPopup.setLoading(true);
+  api
+    .addCard(data)
+    .then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => addCardPopup.setLoading(false, "Create"));
 }
 
+addCardOpenButton.addEventListener("click", () => {
+  formValidators["edit-profile-form"].resetValidation();
+  addCardPopup.open();
+});
+
+/* ------------------------------ EDIT PROFILE ------------------------------ */
+const editProfilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+editProfilePopup.setEventListeners();
+
 function handleProfileEditSubmit(data) {
-  newUserInfo.setUserInfo(data);
-  editProfilePopup.close();
+  editProfilePopup.setLoading(true);
+  api
+    .updateUserInfo(data.name, data.about)
+    .then((res) => {
+      newUserInfo.setUserInfo(res);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => editProfilePopup.setLoading(false, "Save"));
 }
 
-// Event Listeners //
-
 profileEditOpenButton.addEventListener("click", () => {
   const data = newUserInfo.getUserInfo();
   profileTitleInput.value = data.name;
-  profileDescriptionInput.value = data.description;
+  profileDescriptionInput.value = data.about;
   editProfilePopup.open();
 });
 
-addCardOpenButton.addEventListener("click", () => {
-  addCardFormValidator.resetValidation();
-  addCardPopup.open();
+/* -------------------------- UPDATE PROFILE AVATAR ------------------------- */
+const updateAvatarForm = new PopupWithForm(
+  "#avatar-image-modal",
+  handleAvatarFormSubmit
+);
+updateAvatarForm.setEventListeners();
+
+function handleAvatarFormSubmit(data) {
+  updateAvatarForm.setLoading(true);
+  api
+    .updateAvatar(data.avatar)
+    .then((res) => {
+      newUserInfo.setUserAvatar(res.avatar);
+      updateAvatarForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      updateAvatarForm.setLoading(false, "Save");
+    });
+}
+
+editAvatarOpenButton.addEventListener("click", () => {
+  formValidators["edit-avatar-form"].resetValidation();
+  updateAvatarForm.open();
 });
+
+/* ------------------------------- IMAGE POPUP ------------------------------ */
+const imagePreview = new PopupWithImage("#image-modal");
+imagePreview.setEventListeners();
+
+/* -------------------------------- LIKE A CARD ------------------------------- */
+function handleLikeClick(item) {
+  if (!item.isLiked) {
+    api
+      .likeCard(item.getId())
+      .then((res) => {
+        item.updateLikeStatus(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .unlikeCard(item.getId())
+      .then((res) => {
+        item.updateLikeStatus(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors